pkg/storage/pod.go

```diff
@@ -147,15 +140,26 @@ func (s *podStorage) Store(newPods *MetricsBatch) {
 		}
 		containerPoints := len(newPrevPod.Containers)
 		if containerPoints > 0 {
-			prevPods[podRef] = newPrevPod
+			if s.prev == nil {
+				s.prev = make(map[apitypes.NamespacedName]PodMetricsPoint)
```

Comment:
Let's allocate the map completely if that hasn't been done before.

```suggestion
				s.prev = make(map[apitypes.NamespacedName]PodMetricsPoint, len(newPods.Pods))
```